lib/postrank-uri.rb

```diff
@@ -89,6 +89,7 @@ module URI
 
     URIREGEX[:escape]   = /([^ a-zA-Z0-9_.-]+)/x
     URIREGEX[:unescape] = /((?:%[0-9a-fA-F]{2})+)/x
+    URIREGEX[:reserved] = /(%(?:21|23|24|25|26|27|28|29|2A|2B|2C|2F|3A|3B|3D|3F|40|5B|5D))/x
```

Comment:
Is this list the characters from here? https://tools.ietf.org/html/rfc3986#section-2.2

If so, it would be nice to have a way of building this list in a way that it's clear what characters are in here so we can validate it against that. Like have a list of reserved characters and turn them into hex when building the regex.

Or maybe there is a list of hex codes that we can compare this list against?